### ISSUE TITLE
[Bug]: ComposedModal does not close on `esc`

### ISSUE THREAD

**Issue by tay1orjones** [carbon-design-system/carbon#19850]
### Package

@carbon/react

### Browser

Chrome

### Package version

1.86.0

### React version

19

### Description

ComposedModal does not close when <kbd>esc</kbd> is pressed

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-composedmodal--with-state-manager

### Steps to reproduce

Open the modal, press <kbd>esc</kbd>, nothing happens.

### Suggested Severity

Severity 2 = User cannot complete task, and/or no workaround within the user experience of a given component.

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
